lib/deimos/utils/db_poller/base.rb

```diff
@@ -98,13 +120,16 @@ def process_updates
         # @return [Boolean]
         def process_batch_with_span(batch, status)
           retries = 0
+          span = nil
           begin
-            span = Deimos.config.tracer&.start(
-              'deimos-db-poller',
-              resource: @producer.class.name.gsub('::', '-')
-            )
-            process_batch(batch)
-            Deimos.config.tracer&.finish(span)
+            @producers.each do |producer|
+              span = Deimos.config.tracer&.start(
+                'deimos-db-poller',
+                resource: producer.class.name.gsub('::', '-')
+              )
+              process_batch(producer, batch)
+              Deimos.config.tracer&.finish(span)
```

Comment:
I think you should have just one span - it should be set to the poller class rather than the producer class if the poller class is a subclass. Something like 

```ruby
resource_class = self.producers.any? ? self : @producer 
```

Probably best to set this resource_class during initialization since you're in the middle of that logic then anyway.

Reply:
@dorner I noticed you asked me to remove the @producer instance variable above, but I see reference to it on this comment. Should we still keep that variable?

Reply:
No, you don't need it - I mean that when you're looking at setting up the producers, you can also set the resource\_class there as well.

Reply:
Ah okay I see. So instead of storing the @producer variable, I can just look at the config.producer_class directly 👍